src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/RestoreArgs.cs

```diff
@@ -212,7 +212,7 @@ public void ApplyStandardProperties(RestoreRequest request)
                 request.ValidateRuntimeAssets = ValidateRuntimeAssets.Value;
             }
 
-            request.AllowNoOp = AllowNoOp;
+            request.AllowNoOp = request.CacheContext.NoCache ? false : AllowNoOp;
```

Comment:
Seems weird to mention the `false` literal. `!request.CacheContext.NoCache && AllowNoOp`?